### PR TITLE
feat: update script to export final data before embedding

### PR DESCRIPTION
**Pull request by ces-vinhhuynh**
CU-86cu1gaf2

### CODE REVIEW

File: scripts/src/main.ts [hunk 1]
```diff
@@ -18,8 +29,13 @@ if (!existsSync(PLAINTEXT_OUT_PATH)) {
   await mkdir(PLAINTEXT_OUT_PATH);
 }
 
+if (!existsSync(FINAL_OUT_PATH)) {
+  await mkdir(FINAL_OUT_PATH);
+}
+
 try {
   let countPage = 0;
```
[the remark this answers is not in the document]
Comment (ces-vinhhuynh) [reply scripts/src/main.ts]: I updated

File: scripts/src/main.ts [hunk 2]
```diff
@@ -18,8 +29,13 @@ if (!existsSync(PLAINTEXT_OUT_PATH)) {
...
+  const pages = [];
```
[the remark this answers is not in the document]
Comment (ces-vinhhuynh) [reply scripts/src/main.ts]: I intended to include it with the other PR, but I updated

File: scripts/src/main.ts [hunk 3]
```diff
@@ -48,11 +64,22 @@ try {
...
           `${bookData.slug}_${bookContent.slug}`
         );
 
+        const newPage = {
+          id: bookContent.id,
+          name: bookContent.name,
+          url: bookContent.url,
+          description: sanitizedText,
+        };
+        pages.push(newPage);
```
[the remark this answers is not in the document]
Comment (ces-vinhhuynh) [reply scripts/src/main.ts]: I updated

File: scripts/src/utils.ts [hunk 4]
```diff
@@ -30,5 +34,24 @@ export const writePlaintextToOutDir = (value: any, name: string) => {
...
+  }
+
+  return writeFile(
+    path.join(FINAL_OUT_PATH, fileName),
+    JSON.stringify(value, null, 2),
+    {
+      encoding: "utf8",
+    }
+  );
+};
```
[the remark this answers is not in the document]
Comment (ces-vinhhuynh) [reply scripts/src/utils.ts]: I updated

File: scripts/src/generate.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            description,
+          });
+        });
+      } else {
+        throw new Error("Undefined book content type");
+      }
+    });
+  });
+
+  writeFinalOutputToOutDir(pages, "finalOutput");
```
Comment (ces-sonnguyen) [inline scripts/src/generate.ts:85]: "finalOutput" should be a constant or enum

File: scripts/src/main.ts [hunk 6]
```diff
@@ -77,12 +113,12 @@ try {
       }
     }
 
-    countPage += countPagesOfEachBooks;
     console.info(`${book.name} has ${countPagesOfEachBooks} pages`);
     console.info("----------------------------------------");
   }
 
-  console.info(`Total pages in wiki: ${countPage}`);
+  await writeFinalOutputToOutDir(pages, "finalOutput");
```
Comment (ces-sonnguyen) [inline scripts/src/main.ts:120]: same as above

Comment (ces-vinhhuynh) [reply scripts/src/main.ts]: @ces-sonnguyen I updated

**Pull request merged**